### PR TITLE
bpo-15542: Documentation incorrectly suggests __init__ called after direct  __new__ call

### PR DESCRIPTION
**Pull request by nanjekyejoannah**
Co-Authored by : Aaron.Staley

Clarify data model documentation.

cc @cjerdonek 

https://bugs.python.org/issue15542

### CODE REVIEW

**Comment by nanjekyejoannah** [comment]
@willingc , help me apply the skip news entry on this PR.

File: Doc/reference/datamodel.rst [hunk 1]
```diff
@@ -1166,10 +1166,10 @@ Basic customization
    with appropriate arguments and then modifying the newly-created instance
    as necessary before returning it.
 
-   If :meth:`__new__` returns an instance of *cls*, then the new instance's
-   :meth:`__init__` method will be invoked like ``__init__(self[, ...])``, where
-   *self* is the new instance and the remaining arguments are the same as were
-   passed to :meth:`__new__`.
+   If :meth:`__new__` is invoked during object construction and it returns an
+   instance of *cls*, then the new instance’s :meth:`__init__` method will be
```
Comment (rhettinger) [inline Doc/reference/datamodel.rst]: Substitute "instance or subclass of cls" for "instance of cls"

**Review by rhettinger — CHANGES_REQUESTED** [review]

**Comment by nanjekyejoannah** [comment]
I have made the requested changes; please review again .

**Review by willingc — APPROVED** [review]
Looks as if the requested changes are applied. Thanks @nanjekyejoannah and Raymond.

**Review by rhettinger — APPROVED** [review]

**Pull request merged**